Sources/DiscordModels/Types/Gateway.swift

```diff
@@ -469,41 +469,53 @@ public struct Gateway: Sendable, Codable {
         
         /// https://discord.com/developers/docs/topics/gateway-events#identify-identify-connection-properties
         public struct ConnectionProperties: Sendable, Codable {
-            public var os: String = {
-#if os(macOS)
-                return "macOS"
-#elseif os(Linux)
-                return "Linux"
-#elseif os(iOS)
-                return "iOS"
-#elseif os(watchOS)
-                return "watchOS"
-#elseif os(tvOS)
-                return "tvOS"
-#elseif os(Windows)
-                return "Windows"
-#elseif canImport(Musl)
-                return "Musl"
-#elseif os(FreeBSD)
-                return "FreeBSD"
-#elseif os(OpenBSD)
-                return "OpenBSD"
-#elseif os(Android)
-                return "Android"
-#elseif os(PS4)
-                return "PS4"
-#elseif os(Cygwin)
-                return "Cygwin"
-#elseif os(Haiku)
-                return "Haiku"
-#elseif os(WASI)
-                return "WASI"
-#else
-                return "Unknown"
-#endif
-            }()
-            public var browser = "DiscordBM"
-            public var device = "DiscordBM"
+            public var os: String
+            public var browser: String
+            public var device: String
+
+            public init(
+                os: String? = nil,
+                browser: String = "DiscordBM",
+                device: String = "DiscordBM"
+            ) {
+                self.os = os ?? ConnectionProperties.defaultOS()
+                self.browser = browser
+                self.device = device
+            }
+
+            private static func defaultOS() -> String {
```

Comment:
Looks fine but let's be more explicit with the parameter type since we can:
```suggestion
            public init(
                os: String = Self.defaultOS(),
                browser: String = "DiscordBM",
                device: String = "DiscordBM"
            ) {
                self.os = os
                self.browser = browser
                self.device = device
            }

            @usableFromInline
            static func defaultOS() -> String {
```